AutoFinance.Broker/InteractiveBrokers/Controllers/TwsControllerBase.cs

```diff
@@ -849,55 +858,60 @@ public Task<List<PositionStatusEventArgs>> RequestPositions(CancellationToken ca
         /// Set the type for the market data feed
         /// </summary>
         /// <param name="marketDataTypeId">The feed level</param>
-        public void RequestMarketDataType(int marketDataTypeId)
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public Task RequestMarketDataTypeAsync(int marketDataTypeId)
         {
-            this.clientSocket.RequestMarketDataType(marketDataTypeId);
+            // Set the operation to cancel after 5 seconds
+            CancellationTokenSource tokenSource = new CancellationTokenSource(5000);
+            return this.RequestMarketDataTypeAsync(marketDataTypeId, tokenSource.Token);
         }
 
         /// <summary>
-        /// Get the PnL of the account.
+        /// Set the type for the market data feed
         /// </summary>
-        /// <param name="accountCode">The account code</param>
-        /// <param name="modelCode">The model code</param>
-        /// <returns>The PnL account update event from TWS.</returns>
-        public Task<PnLEventArgs> RequestPnL(
-            string accountCode,
-            string modelCode)
+        /// <param name="marketDataTypeId">The feed level</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public Task RequestMarketDataTypeAsync(int marketDataTypeId, CancellationToken cancellationToken)
```

Comment:
Make an integration test around this method

Reply:
Done.